lib/models/application.coffee

```diff
@@ -446,6 +446,95 @@ getApplicationModel = (deps, opts) ->
 		.asCallback(callback)
 
 	###*
+	# @summary Purge devices by application id
+	# @name purge
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {Number} appId - application id
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.purge(123);
+	#
+	###
+	exports.purge = (appId) ->
+		return request.send
+			method: 'POST'
+			url: '/supervisor/v1/purge'
+			baseUrl: apiUrl
+			body:
+				appId: appId
+				data:
+					appId: "#{appId}"
+		.catch (err) ->
+			if err.statusCode == LOCKED_STATUS_CODE
+				throw new errors.ResinSupervisorLockedError()
+
+			throw err
+
+	###*
+	# @summary Shutdown devices by application id
+	# @name shutdown
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {Number} appId - application id
+	# @param {Object} [options] - options
+	# @param {Boolean} [options.force=false] - override update lock
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.shutdown(123);
+	###
+	exports.shutdown = (appId, options = {}) ->
+		return request.send
+			method: 'POST'
+			url: '/supervisor/v1/shutdown'
+			baseUrl: apiUrl
+			body:
+				appId: appId
+				data:
+					force: Boolean(options.force)
+		.catch (err) ->
+			if err.statusCode == LOCKED_STATUS_CODE
+				throw new errors.ResinSupervisorLockedError()
+
+			throw err
+
+	###*
+	# @summary Reboot devices by application id
+	# @name reboot
+	# @public
+	# @function
+	# @memberof resin.models.application
+	#
+	# @param {Number} appId - application id
+	# @param {Object} [options] - options
+	# @param {Boolean} [options.force=false] - override update lock
+	# @returns {Promise}
+	#
+	# @example
+	# resin.models.application.reboot(123);
+	###
+	exports.reboot = (appId, options = {}) ->
+		return request.send
+			method: 'POST'
+			url: '/supervisor/v1/reboot'
+			baseUrl: apiUrl
+			body:
+				appId: appId
+				data:
+					force: Boolean(options.force)
+		.catch (err) ->
+			if err.statusCode == LOCKED_STATUS_CODE
+				throw new errors.ResinSupervisorLockedError()
+
+			throw err
+
```

Comment:
Can you have these new application methods accept and optional callback, the same way that the device methods do?